AK/Utf16View.h

```diff
@@ -456,31 +456,24 @@ class Utf16View {
 
     [[nodiscard]] constexpr bool starts_with(Utf16View const& needle) const
     {
-        if (needle.is_empty())
+        auto needle_length = needle.length_in_code_units();
+        if (needle_length == 0)
             return true;
-        if (is_empty())
-            return false;
-        if (needle.length_in_code_units() > length_in_code_units())
+        if (is_empty() || needle.length_in_code_units() > length_in_code_units())
             return false;
 
-        if (has_ascii_storage() && needle.has_ascii_storage()) {
-            if (m_string.ascii == needle.m_string.ascii)
-                return true;
-            return ascii_span().starts_with(needle.ascii_span());
-        }
-
-        if (!has_ascii_storage() && !needle.has_ascii_storage()) {
-            if (m_string.utf16 == needle.m_string.utf16)
-                return true;
-            return utf16_span().starts_with(needle.utf16_span());
-        }
+        return substring_view(0, needle_length) == needle;
```

Comment:
This does lose an optimization where we checked if the pointers were the same after checking the length. That was based on:
https://github.com/LadybirdBrowser/ladybird/blob/265e278275f699859092bcb2743c083591678ff2/AK/Utf8View.cpp#L117-L118

I don't know if that was particularly significant though, because it appears other string APIs don't do this either, and it was just part of the initial commit to add `Utf8View::starts_with` (edecf8f6a375b04e60e7c4ddcc10b40f350a5d77).

Reply:
Oh, there is a nice way to get that back: we can reintroduce the pointer check in `TypedTransfer<T>::compare()`, which does not need to perform any comparisons if it was passed two identical pointers.